File: src/main/java/logan/pickpocket/user/WorldGuardUtil.java
package logan.pickpocket.user;

import com.sk89q.worldedit.bukkit.BukkitAdapter;
import com.sk89q.worldguard.WorldGuard;
import com.sk89q.worldguard.bukkit.WorldGuardPlugin;
import com.sk89q.worldguard.protection.regions.RegionContainer;
import com.sk89q.worldguard.protection.regions.RegionQuery;
import logan.pickpocket.main.PickpocketPlugin;
import org.bukkit.entity.Player;

public final class WorldGuardUtil {

    private WorldGuardUtil() {
    }

    public static boolean isPickpocketingAllowed(Player player) {
        if (!PickpocketPlugin.isWorldGuardPresent()) return true;
        var localPlayer = WorldGuardPlugin.inst().wrapPlayer(player);
        RegionContainer container = WorldGuard.getInstance().getPlatform().getRegionContainer();
        RegionQuery query = container.createQuery();
        var worldEditLocation = BukkitAdapter.adapt(player.getLocation());
        return query.testState(worldEditLocation, localPlayer, PickpocketPlugin.getPickpocketFlag());
    }
}

File: src/main/java/logan/pickpocket/user/WorldGuardUtil.kt
package logan.pickpocket.user

import com.sk89q.worldedit.bukkit.BukkitAdapter
import com.sk89q.worldguard.WorldGuard
import com.sk89q.worldguard.bukkit.WorldGuardPlugin
import logan.pickpocket.main.PickpocketPlugin
import org.bukkit.entity.Player

class WorldGuardUtil {
    companion object {
        fun isPickpocketingAllowed(player: Player): Boolean {
            if (!PickpocketPlugin.isWorldGuardPresent) return true
            val localPlayer = WorldGuardPlugin.inst().wrapPlayer(player)
            val container = WorldGuard.getInstance().platform.regionContainer
            val query = container.createQuery()
            val worldEditLocation = BukkitAdapter.adapt(player.location)
            return query.testState(worldEditLocation, localPlayer, PickpocketPlugin.PICKPOCKET_FLAG)
        }
    }
}